spec/schemas/register_schema.json

```diff
@@ -0,0 +1,157 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+
+  "title": "Register File Schema",
+  "description": "Schema for describing a register file",
+
+  "$defs": {
+    "count_variant": {
+      "type": "object",
+      "required": ["value", "when"],
+      "additionalProperties": false,
+      "properties": {
+        "value": {
+          "type": "integer",
+          "description": "Override count when condition is satisfied"
+        },
+        "when": {
+          "$ref": "schema_defs.json#/$defs/requires_entry"
+        },
+        "description": {
+          "$ref": "schema_defs.json#/$defs/spec_text"
+        }
+      }
+    },
+    "register_entry": {
+      "type": "object",
+      "required": ["name", "index"],
+      "additionalProperties": false,
+      "properties": {
+        "name": {
+          "$ref": "schema_defs.json#/$defs/register_name"
+        },
+        "abi_mnemonic": {
+          "$ref": "schema_defs.json#/$defs/register_alias"
+        },
+        "index": {
+          "type": "integer",
+          "minimum": 0,
+          "description": "Register index"
+        },
+        "description": {
+          "$ref": "schema_defs.json#/$defs/spec_text"
+        },
+        "definedBy": {
+          "$ref": "schema_defs.json#/$defs/requires_entry"
+        },
+        "when": {
+          "$ref": "schema_defs.json#/$defs/requires_entry"
+        },
+        "length": {
+          "$ref": "schema_defs.json#/$defs/bit_length_value"
+        },
```

Comment:
I'm not sure we want to allow this. Having a register file where each entry can have its own length might make the code which uses them harder to write correctly. Maybe we remove this unless and until it's needed?